core/ErrorQueue.h

```diff
@@ -18,7 +18,7 @@ class ErrorQueue {
     std::vector<std::unique_ptr<ErrorQueueMessage>> drainFlushed();
     void collectForFile(core::FileRef whatFile, std::vector<std::unique_ptr<core::ErrorQueueMessage>> &out);
     ErrorFlusher errorFlusher;
-    const std::thread::id owner;
```

Comment:
see suggestion in another comment of re-creating error queue instead of making this field mutable.
I'd strongly prefer that.